src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -386,17 +384,41 @@ private AddInterfaceInteraction createAddInterfaceInteractionAction(@Nonnull Pay
         CompletionStage<PaymentHandleResponse> paymentHandleResponseStage = ctpPaymentIdOptStage
                 .thenCompose(ctpPaymentIdOpt ->
                         ctpPaymentIdOpt.map(ctpPaymentId -> {
-                            AddInterfaceInteraction action = createAddInterfaceInteractionAction(restException.getDetails(), RESPONSE);
-                            return ctpFacade.getPaymentService().updatePayment(ctpPaymentId, Collections.singletonList(action))
-                                    .thenApply(ignore -> PaymentHandleResponse.ofHttpStatusAndErrorMessage(HttpStatus.valueOf(restException.getResponsecode()),
-                                            format("%s=[%s] can't be processed, details: [%s]", paymentIdType, paymentId, restException.getMessage())));
+                            return updatePaymentResponse(paymentId, paymentIdType, throwable, ctpPaymentId);
                         }).orElseGet(() -> CompletableFuture.completedFuture(
                                 PaymentHandleResponse.of404NotFound(format("%s=[%s] is not found.", paymentIdType, paymentId))
                         ))
                 );
         return paymentHandleResponseStage;
     }
 
+    private CompletionStage<PaymentHandleResponse> updatePaymentResponse(@Nullable String paymentId, @Nullable String paymentIdType,
+                                                                         @Nonnull Throwable throwable, String ctpPaymentId) {
+
+        if (throwable instanceof PaypalPlusServiceException) {
```

Comment:
Do you have tests for this part?

Reply:
It is not possible to reproduce this unpredictable erroneous situation with IT nor unit tests. This block of code only runs when there is an unknown exception which will only redirects back to the API caller.

Reply:
Can you clarify when these error cases can happen?

Reply:
Added integration tests to cover both the exceptional scenarios b1eab7b8cc903ed48eaf951e8a8c6f9e167e1f8e